### PR TITLE
Refactor calendar and increase event description max

### LINKED ISSUE #131: Show more of the description for events
it would be nice if in this example both links would be shown
image

Possible fixes:
- show more of the description
- shrink links drastically (eg display only domain)

### PR DESCRIPTION
**Pull request by DenverCoder1**
Removed unnecessary functions

Changed `Calendar` and `Event` to use property and static factory methods

Several improvements for readability

Increased trim size on descriptions (Closes #131)

### CODE REVIEW

File: modules/calendar/event.py [hunk 1]
```diff
@@ -4,78 +4,122 @@
...
+		self.__title = title
+		self.__location = location
+		self.__description = description
+		self.__all_day = all_day
+		self.__start = start
+		self.__end = end
+		self.__timezone = timezone
+
+	@property
 	def id(self) -> str:
```
Comment (abrahammurciano) [inline modules/calendar/event.py:32]:
```suggestion
	def event_id(self) -> str:
```
better to not use same names as built in functions

File: modules/calendar/event.py [hunk 2]
```diff
@@ -4,78 +4,122 @@
...
+	def __relative_end_str(self, base=datetime.now()) -> str:
+		"""Returns a formatted string of the end date"""
+		end_date = self.end
+		# use previous day if end of multi-day, all-day event
+		if self.all_day and not self.__one_day:
+			end_date -= timedelta(days=1)
+		return format_date(end_date, all_day=self.all_day, base=base)
+
+	@classmethod
+	def from_dict(cls, details: Dict[str, str]) -> "Event":
```
Comment (abrahammurciano) [inline modules/calendar/event.py:104]:
```suggestion
	def from_dict(cls, details: Dict[str, Any]) -> "Event":
```
could be nested dicts, not necessarily str

**Review by abrahammurciano — APPROVED** [review]

File: modules/calendar/event.py (hunk 1 above)
Comment (DenverCoder1) [reply modules/calendar/event.py:32]: Objects do not have an `id` property unless it is specifically declared, so this does not cause any confusion.

Comment (DenverCoder1) [reply modules/calendar/event.py:32]: Even the Discord API uses id properties in classes

**Review by abrahammurciano — APPROVED** [review]

**Pull request merged**